src/cmd/run.go

```diff
@@ -46,6 +53,11 @@ func doRun(cmd *cobra.Command, args []string) {
 	runner, err := client.RunnerRegister()
 	pkg.CheckErr(err)
 
+	if viper.GetBool("scaling-enabled") {
+		log.Info().Msgf("electing leader...")
+		electLeader()
```

Comment:
So it my understanding this function will forever block because eventually it will get to `pkg.RunLeaderElection` which has `leaderelection.RunOrDie` in it which my understanding is its forever blocking untill the `ctx` is cancelled.  This means the code will never progress past here when `--scaling-enabled=true`.

Is this understanding correct?